perf(codegen): optimize `Codegen::print_list` method

Dunqing opened the pull request:
It looks like has fewer instructions than before. See https://godbolt.org/z/jE1o5rvGq

NOTE: This approach didn't bring notable performance improvement, so I don't have a strong opinion on which way.

Dunqing commented:
* **#10242** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10242?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10242?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=10242).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen commented:
Look at the comment above, but seems like it's better this time?

Dunqing commented:
> Look at the comment above this function, seems like it's better this time?

It seems better than the [last optimization](https://github.com/oxc-project/oxc/pull/5221) while maintaining readability.

Dunqing commented:
It seems adding `#[inline]` has no impact on performance.

Boshen commented:
### Merge activity

* **Apr 7, 3:35 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Dunqing committed:
perf(codegen): optimize `Codegen::print_list` method (#10242)

It looks like has fewer instructions than before. See https://godbolt.org/z/jE1o5rvGq

NOTE: This approach didn't bring notable performance improvement, so I don't have a strong opinion on which way.

Dunqing commented:
> Just to clarify, the purpose of this change is better readability? From the [Godbolt link](https://godbolt.org/z/jE1o5rvGq) you gave, it looks like it makes no difference to assembly either way.
> 
> That's not criticism, I'm just interested.

Yes, the performance is unchanged. Boshen said that merging it because better readability.
